gossip/comm/comm_test.go

```diff
@@ -624,35 +624,56 @@ func TestCommSend(t *testing.T) {
 		}
 	}()
 
-	c1recieved := 0
-	c2recieved := 0
+	c1received := 0
+	c2received := 0
 	// hopefully in some runs we'll fill both send and receive buffers and
 	// drop overflowing messages, but still finish, because the endless
 	// stream of messages inexorably gets through unless something is very
 	// broken.
-	totalMessagesRecieved := (DefSendBuffSize + DefRecvBuffSize) * 2
-RECV:
-	for {
-		select {
-		case <-ch1:
-			c1recieved++
-			if c1recieved == totalMessagesRecieved {
-				close(stopch2)
-			}
-		case <-ch2:
-			c2recieved++
-			if c2recieved == totalMessagesRecieved {
-				close(stopch1)
+	totalMessagesReceived := (DefSendBuffSize + DefRecvBuffSize) * 2
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	// read from each until there are an acceptable number of
+	// messages and then stop the sender from sending more
+	// messages.
+	go func() {
+		for range ch1 {
+			c1received++
+			if c1received == totalMessagesReceived {
+				wg.Done()
+				return
 			}
-		case <-time.After(30 * time.Second):
-			t.Fatalf("timed out waiting for messages to be recieved.\nc1 got %d messages\nc2 got %d messages", c1recieved, c2recieved)
-		default:
-			if c1recieved > totalMessagesRecieved && c2recieved > totalMessagesRecieved {
-				break RECV
+		}
+	}()
+
+	go func() {
+		for range ch2 {
+			c2received++
+			if c2received == totalMessagesReceived {
+				wg.Done()
+				return
 			}
 		}
+	}()
+
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		wg.Wait()
+		close(stopch1)
+		close(stopch2)
+	}()
+
+	ticker := time.NewTicker(30 * time.Second)
+	select {
+	case <-c:
+	case <-ticker.C:
```

Comment:
why not use **time.After** ? This is done one time, so who cares if the ticker is still scheduled to run?